Runtime/Core/ProBuilderMeshFunction.cs

```diff
@@ -376,9 +382,32 @@ public void ToMesh(MeshTopology preferredTopology = MeshTopology.Triangles)
                 if (submeshes[i] == null)
                     throw new Exception("Attempting to assign a null submesh. " + i + "/" + materialCount);
 #endif
-                mesh.SetIndices(submeshes[i].m_Indexes, submeshes[i].m_Topology, i, false);
+                if (submeshes[i].m_Indexes.Length == 0)
+                {
+                    m_MeshMaterials.RemoveAt(submeshes[i].submeshIndex);
+
+                    foreach (var face in facesInternal)
+                    {
+                        if (face.submeshIndex == submeshes[i].submeshIndex + 1)
+                            face.submeshIndex = submeshes[i].submeshIndex;
```

Comment:
How does this handle the case where there are multiple materials at a higher index than the removed? E.x., consider the case where there are 5 materials in use, and you remove the first two. Does this code correctly shift the remaining indices?

Reply:
I've fixed it in [3772d72](https://github.com/Unity-Technologies/com.unity.probuilder/pull/460/commits/3772d72255c8887841c75887d34a1c1d4e104775)